devito/operator.py

```diff
@@ -190,14 +190,23 @@ def arguments(self, *args, **kwargs):
         # Insert loop size arguments from dimension values
         d_args = [d for d in arguments.values() if isinstance(d, Dimension)]
         for d in d_args:
-            arguments[d.name] = dim_sizes[d]
+            if d not in dim_sizes and d.is_Buffered:
+                # Only the parent is defined, so use it
+                arguments[d.name] = dim_sizes[d.parent]
+            else:
+                arguments[d.name] = dim_sizes[d]
```

Comment:
Ahhh ok, I think I understand now. Since it's a function call now, `d.parent` always gets evaluated, but that obviously fails in many cases. Even better though, removing the change entirely seems to have fixed the problem now (post-rebase), most likely due to the other bug-fix in a follow-up commit. I'll drop this for now and see if travis agrees.